pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,129 @@
+            super().__init__()
+            self.interpolation_mode = interpolation_mode
+
+        def forward(
+            self, image: torch.Tensor, maximal_size: int = 1800, minimal_size: int = 800
+        ) -> torch.Tensor:
+            if max(image.shape) > maximal_size:
+                return F.rescale(
+                    image, factor=maximal_size / max(extract_image_size(image))
+                )
+            if min(image.shape) < minimal_size:
+                alpha = minimal_size / float(min(extract_image_size(image)))
+                if alpha < 4.0:
+                    return F.rescale(image, factor=alpha)
+                else:
+                    return cast(
+                        torch.Tensor,
+                        F.resize(
+                            image, (minimal_size, minimal_size), self.interpolation_mode
+                        ),
+                    )
+            return image
+
+        def _properties(self) -> Dict[str, Any]:
+            dct = super()._properties()
+            if self.interpolation_mode != "bilinear":
+                dct["interpolation_mode"] = self.interpolation_mode
+            return dct
+
+    transforms = (
+        OptionalRescale(),
+        ValidRandomCrop((edge_size, edge_size)),
+        OptionalGrayscaleToFakegrayscale(),
+    )
+    return ComposedTransform(*transforms)
+
+
+# TODO: Implement this
+# def images(root: Optional[str] = None,
+#   download: bool = True,
+#   overwrite: bool = False):
+#
+#     # base_sanakoyeu =
+#     "https://hcicloud.iwr.uni-heidelberg.de/index.php/s/NcJj2oLBTYuT1tf/download?path=%2F&files="
+#     # tar1 = "berthe-morisot.tar.gz"
+#     # tar2 = "edvard-munch.tar.gz"
+#     # tar3 = "el-greco.tar.gz"
+#     # tar4 = "ernst-ludwig-kirchner.tar.gz"
+#     # tar5 = "jackson-pollock.tar.gz"
+#     # tar6 = "monet_water-lilies-1914.tar.gz"
+#     # tar7 = "nicholas-roerich.tar.gz"
+#     # tar8 = "pablo-picasso.tar.gz"
+#     # tar9 = "paul-cezanne.tar.gz"
+#     # tar10 = "paul-gauguin.tar.gz"
+#     # tar11 = "sample_photographs.tar.gz"
+#     # tar12 = "samuel-peploe.tar.gz"
+#     # tar13 = "vincent-van-gogh_road-with-cypresses-1890.tar.gz"
+#     # tar14 = "wassily-kandinsky.tar.gz"
+#     # places365_url = (
+#     #     "data.csail.mit.edu/places/places365/train_large_places365standard.tar"
+#     # )
+#
+#     return None
+
+
+def dataset(root: str, transform: Optional[Transform] = None,) -> ImageFolderDataset:
+    if transform is None:
+        transform = image_transform()
+    return ImageFolderDataset(root, transform=transform)
+
+
+def batch_sampler(
+    data_source: Sized,
+    impl_params: bool = True,
+    num_batches: Optional[int] = None,
+    batch_size: Optional[int] = None,
+) -> FiniteCycleBatchSampler:
+
+    if num_batches is None:
+        # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/main.py#L68
+        num_batches = 300_000 if impl_params else 100_000
```

Comment:
I recall something related to this. I think we split the `num_batches` in multiple epochs? Do you remember? :smile:

Reply:
That's right. The paper describes that the `lr` is reduced by a factor of 10 after 200k steps.

Reply:
According to this without `impl_params` we only have 100k steps. Thus, both can't be true. Which one is it?